pkg/watch/restricted_enqueue_request_for_owner.go

```diff
@@ -1,227 +0,0 @@
-)
-
-const (
-	Status = "status"
-	State  = "state"
-)
-
-var (
-	ErrNoUniqueKind = errors.New("multiple kind candidates are invalid")
-	ErrStateInvalid = errors.New("state from component object could not be interpreted")
-)
-
-// EnqueueRequestForOwner enqueues Requests for the Owners of an object.  E.g. the object that created
-// the object that was the source of the Event.
-//
-// If a ReplicaSet creates Pods, users may reconcile the ReplicaSet in response to Pod Events using:
-//
-// - a source.Kind Source with Type of Pod.
-//
-// - a handler.EnqueueRequestForOwner EventHandler with an OwnerType of ReplicaSet and IsController set to true.
-type RestrictedEnqueueRequestForOwner struct {
-	Log logr.Logger
-
-	// OwnerType is the type of the Owner object to look for in OwnerReferences.  Only Group and Kind are compared.
-	OwnerType machineryruntime.Object
-
-	// IsController if set will only look at the first OwnerReference with Controller: true.
-	IsController bool
-
-	// groupKind is the cached Group and Kind from OwnerType
-	groupKind schema.GroupKind
-
-	// mapper maps GroupVersionKinds to Resources
-	mapper meta.RESTMapper
-}
-
-// Create implements EventHandler.
-func (e *RestrictedEnqueueRequestForOwner) Create(
-	_ context.Context,
-	evt event.CreateEvent,
-	q workqueue.TypedRateLimitingInterface[ctrl.Request],
-) {
-	reqs := map[reconcile.Request]any{}
-	e.getOwnerReconcileRequest(nil, evt.Object, reqs)
-	for req := range reqs {
-		q.Add(req)
-	}
-}
-
-// Update implements EventHandler.
-func (e *RestrictedEnqueueRequestForOwner) Update(
-	_ context.Context,
-	evt event.UpdateEvent,
-	q workqueue.TypedRateLimitingInterface[ctrl.Request],
-) {
-	reqs := map[reconcile.Request]any{}
-	e.getOwnerReconcileRequest(evt.ObjectOld, evt.ObjectNew, reqs)
-	for req := range reqs {
-		q.Add(req)
-	}
-}
-
-// Delete implements EventHandler.
-func (e *RestrictedEnqueueRequestForOwner) Delete(
-	_ context.Context,
-	evt event.DeleteEvent,
-	q workqueue.TypedRateLimitingInterface[ctrl.Request],
-) {
-	reqs := map[reconcile.Request]any{}
-	e.getOwnerReconcileRequest(nil, evt.Object, reqs)
-	for req := range reqs {
-		q.Add(req)
-	}
-}
-
-// Generic implements EventHandler.
-func (e *RestrictedEnqueueRequestForOwner) Generic(
-	_ context.Context,
-	evt event.GenericEvent,
-	q workqueue.TypedRateLimitingInterface[ctrl.Request],
-) {
-	reqs := map[reconcile.Request]any{}
-	e.getOwnerReconcileRequest(nil, evt.Object, reqs)
-	for req := range reqs {
-		q.Add(req)
-	}
-}
-
-// parseOwnerTypeGroupKind parses the OwnerType into a Group and Kind and caches the result.  Returns false
-// if the OwnerType could not be parsed using the scheme.
-func (e *RestrictedEnqueueRequestForOwner) parseOwnerTypeGroupKind(scheme *machineryruntime.Scheme) error {
-	// Get the kinds of the type
-	kinds, _, err := scheme.ObjectKinds(e.OwnerType)
-	if err != nil {
-		e.Log.Error(err, "Could not get ObjectKinds for OwnerType", "owner type", fmt.Sprintf("%T", e.OwnerType))
```

Comment:
We need to watch out here since the default implementation will [panic if the `OwnerType` parsing is erroring out](https://github.com/kubernetes-sigs/controller-runtime/blob/b901db121e1f53c47ec9f9683fad90a546688c3e/pkg/handler/enqueue_owner.go#L70). Let's keep this in mind when this change goes live.

Reply:
This panic makes sense, if the scheme doesn't contain Kyma GKV, it doesn't make sense to initiate KLM.